src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -179,6 +179,10 @@ void primitive_inst::update_shape() {
     if (input_shape_changed)
         set_shape_change();
 
+    // We assume that tensor ranks are static, thus shape_of doesn't need to update anything even if input shape is dynamic
+    if (_node->is_type<shape_of>() && !input_shape_changed)
```

Comment:
Hmm, wouldn't overloading of `get_shape_infer_dependencies` in shape_of node (which will return empty vector) help to resolve this issue?

Reply:
Yeah, it will fix. But wouldn't that be misleading? Since actually shape_of may depend on input in case of dynamic rank (which is not supported, but still...)